File: benchmark/python/arrays.py
#!/usr/bin/env python3
# arrays.py - Array Operations Benchmark for Python
# Tests dynamic array operations and iteration

import time

def main():
    size = 1000000

    start = time.time()

    # Create array with 1,000,000 integers (0 to 999,999)
    arr = list(range(size))

    # Sum all elements
    sum1 = sum(arr)

    # Reverse the array in-place
    arr.reverse()

    # Sum again to verify
    sum2 = sum(arr)

    elapsed = int((time.time() - start) * 1000)

    print(f"Sum: {sum1}")
    print(f"Reversed sum: {sum2}")
    print(f"Array time: {elapsed}ms")

if __name__ == "__main__":
    main()

File: benchmark/nodejs/arrays.js
#!/usr/bin/env node
// arrays.js - Array Operations Benchmark for Node.js
// Tests dynamic array operations and iteration

function main() {
    const size = 1000000;

    const start = Date.now();

    // Create array with 1,000,000 integers (0 to 999,999)
    const arr = new Array(size);
    for (let i = 0; i < size; i++) {
        arr[i] = i;
    }

    // Sum all elements
    let sum1 = 0n;
    for (let i = 0; i < size; i++) {
        sum1 += BigInt(arr[i]);
    }

    // Reverse the array in-place
    arr.reverse();

    // Sum again to verify
    let sum2 = 0n;
    for (let i = 0; i < size; i++) {
        sum2 += BigInt(arr[i]);
    }

    const elapsed = Date.now() - start;

    console.log(`Sum: ${sum1}`);
    console.log(`Reversed sum: ${sum2}`);
    console.log(`Array time: ${elapsed}ms`);
}

main();
